patches/minecraft/net/minecraft/item/SpawnEggItem.java.patch

```diff
@@ -0,0 +1,82 @@
+--- a/net/minecraft/item/SpawnEggItem.java
++++ b/net/minecraft/item/SpawnEggItem.java
+@@ -35,18 +_,28 @@
+ 
+ public class SpawnEggItem extends Item {
+    private static final Map<EntityType<?>, SpawnEggItem> field_195987_b = Maps.newIdentityHashMap();
++   private static final java.util.List<SpawnEggItem> MOD_EGGS = new java.util.ArrayList<>();
+    private final int field_195988_c;
+    private final int field_195989_d;
+-   private final EntityType<?> field_200890_d;
++   private final java.util.function.Supplier<? extends EntityType<?>> field_200890_d;
+ 
++   /**@deprecated Use {@link SpawnEggItem#SpawnEggItem(java.util.function.Supplier, int, int, Item.Properties)}*/@Deprecated
+    public SpawnEggItem(EntityType<?> p_i48465_1_, int p_i48465_2_, int p_i48465_3_, Item.Properties p_i48465_4_) {
+       super(p_i48465_4_);
+-      this.field_200890_d = p_i48465_1_;
++      this.field_200890_d = p_i48465_1_ == null ? () -> null : p_i48465_1_.delegate; //TODO - 1.17: Remove check for null and enforce it being nonnull
+       this.field_195988_c = p_i48465_2_;
+       this.field_195989_d = p_i48465_3_;
+       field_195987_b.put(p_i48465_1_, this);
+    }
+ 
++   public SpawnEggItem(java.util.function.Supplier<? extends EntityType<?>> typeIn, int primaryColorIn, int secondaryColorIn, Item.Properties builder) {
++      super(builder);
```

Comment:
It might be better/more compatible to delegate to the normal constructor here.

Reply:
That would poison the `BY_ID` map with an entry with a null key and necessitate the null check in the original constructor because I would need to pass null as the type.